grails-app/services/org/pih/warehouse/data/LocationDataService.groovy

```diff
@@ -30,30 +28,42 @@ class LocationDataService {
     Boolean validateData(ImportDataCommand command) {
 
         Locale locale = new Locale(grailsApplication.config.openboxes.locale.defaultLocale)
+        def locationNamesToImport = command.data.collect {it.name}
 
         command.data.eachWithIndex { params, index ->
 
-            Location location = createOrUpdateLocation(params)
-
+            Location newLocation = Location.findByName(params.name)
```

Comment:
I think there is no need for this assignment, and the naming is confusing too. Just use the `Location.findByName(params.name)` in the if condition, where you check if it exists.

Reply:
Done